packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -851,17 +831,28 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
     private attachListener() {
         this.setMaxListeners(Number.MAX_SAFE_INTEGER);
         this.dataStoreContext.once("attaching", () => {
-            assert(this.bindState !== BindState.NotBound,
-                0x186 /* "Data store attaching should not occur if it is not bound" */);
-            this._attachState = AttachState.Attaching;
+            assert(
+                this.localAttachState !== AttachState.Detached,
+                "Data store should not globally attach if it's locally detached",
+            );
+            this._globalAttachState = AttachState.Attaching;
             // This promise resolution will be moved to attached event once we fix the scheduler.
             this.deferredAttached.resolve();
+
+            // Mark all the child contexts as attached.
+            this.localChannelContextQueue.forEach((channel) => {
```

Comment:
This is very important to do because otherwise we will run into this bug - https://github.com/microsoft/FluidFramework/issues/8305.
This is what had happened last time I tried a similar change - https://github.com/microsoft/FluidFramework/pull/8269.